projects/uitdatabank/docs/search-api/advanced-queries.md

```diff
@@ -0,0 +1,1067 @@
+
+**Examples**
+
+Only events that take place in postal code 9000 `AND` that are free too will be returned:
+
+```
+GET /events/?q=address.\*.postalCode:9000 AND price:0
+```
+
+All events that take place in postal code 9000 `OR` the free events that are outside postal code 9000 will be returned:
+
+```
+GET /events/?q=address.\*.postalCode:9000 OR price:0
+```
+
+All events that take place in postal code 9000 will be returned, except (`NOT`) if they are for free:
+
+```
+GET /events/?q=address.\*.postalCode:9000 NOT price:0
+```
+
+## Usage of round brackets
+
+Use round brackets in your advanced queries to:
+
+* group values that belong to the same field
+* specify the order (hierarchy) between different fields
+* avoid the need to repeat fields
+* keep your query simple, clean and more human readable
+
+### Use round brackets to group values in the query
+
+In the example below, no round brackets are used. Because of this, the field `address.\*.postalCode` is repeated 3 times. This is unneccesary and makes the query dificult to read.
+
+```
+GET /events/?q=address.\*.postalCode:9000 OR address.\*.postalCode:1000 OR address.\*.postalCode:2000 OR address.\*.postalCode:3000
+```
+
+Round brackets make it possible to group values and avoid the need to repeat fields. As a result, the query is shorter, more simple and easier to maintain:
+
+```
+GET /events/?q=address.\*.postalCode:(9000 OR 1000 OR 2000 OR 3000)
+```
+
+### Use round brackets to specify the order between different fields
+
+In the case of more complex queries the correct usage of round brackets is mandatory.
+
+Imagine you want to retrieve all concerts (term id `0.50.4.0.0`) and expositions (term id `0.0.0.0.0`) that happen in Ghent (postal code `9000`) or Leuven (postal code `3000`).
+
+Therefore, first we need to make 2 OR relations:
+
+1. between the 2 postal code values (because we want to retrieve events from Ghent *or* Leuven): `address.\*.postalCode:(9000 OR 3000)`
+2. between the 2 term id values (because we want to retrieve concerts *or* expositions): `terms.id:(0.0.0.0.0 OR 0.50.4.0.0)`
+
+Consequently we need to link both OR-relations with an AND operator:
+
+```
+/events/?q=address.\*.postalCode:(9000 OR 3000) AND terms.id:(0.0.0.0.0 OR 0.50.4.0.0)
+```
+
+Now the query will return concerts and expositions that either happen in Ghent or Leuven. Without the usage of round brackets you would get incomplete or incorrect results.
+
+## Escaping
+
+Special characters like `:`, `"` or `*` must always be escaped if used as part of a field name or field value.
+
+### Examples
+
+**Using wildcards in field names**
+
+You can use wildcards in field names to search across multiple translated versions. When doing so, the wildcard must be escaped using a backslash, like in the example below:
+
+```
+GET /events/q=address.\*.postalCode:9000
+```
+
+**Using date time filters**
+
+A colon (`:`) must always be escaped using a backlash (`\`), e.g. in a date time filter:
```

Comment:
As mentioned on Slack I don't think this is 100% correct. For example this works (and is exactly the same as the example below but without the escaping):

```
https://search.uitdatabank.be/events/?disableDefaultFilters=true&q=dateRange:[2022-01-01T00:00:00%2B01:00%20TO%202022-12-31T23:59:59%2B01:00]
```